### PR TITLE
Add ElementByteSizeObserver default constructor

### PR DESCRIPTION
**Pull request by bjchambers**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [*] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [*] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [*] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [*] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

This enables constructing an observer without the Counter to simplify
testing and use with other ways of reporting byte size.

Once compatibility issues are resolved, the class will become abstract
with the Counter-specific functionality removed.

### CODE REVIEW

**Comment by bjchambers** [comment]
R: @tgroh (oneof @dhalperi @kennknowles)

**Comment by tgroh** [comment]
LGTM

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/common/ElementByteSizeObserver.java [hunk 1]
```diff
@@ -32,11 +32,24 @@
   private long totalSize = 0;
   private double scalingFactor = 1.0;
 
+  public ElementByteSizeObserver() {
+    this.counter = null;
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/common/ElementByteSizeObserver.java]: mark `this.counter` `@Nullable`?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java [hunk 2]
```diff
@@ -362,4 +362,65 @@
...
+      throws Exception {
+    TestElementByteSizeObserver observer = new TestElementByteSizeObserver();
+
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    for (T elem : elements) {
+      coder.registerByteSizeObserver(elem, observer, context);
+      coder.encode(elem, os, context);
+      observer.advance();
+    }
+    long expectedLength = os.toByteArray().length;
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java]: Just use `new CountingOutputStream(ByteStreams.nullOutputStream())`?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java [hunk 3]
```diff
@@ -362,4 +362,65 @@
...
+    public long getCount() {
+      return count;
+    }
+
+    public void reset() {
+      currentSum = 0;
+      count = 0;
+    }
+
+    public long getAndReset() {
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java]: `getSumAndReset`?

**Comment by dhalperi** [comment]
Minor comments, but LGTM. Please self-merge.

**Pull request closed without merging**